src/lcancel.c

```diff
@@ -196,35 +197,53 @@ void LCancel_Think(LCancelData *event_data, FighterData *hmn_data)
     JOBJ *hud_jobj = event_data->hud.gobj->hsd_object;
 
     bool should_update_fastfall_hud = false;
+    bool can_fastfall = hmn_data->phys.self_vel.Y < 0;
 
     // log fastfall frame
     // if im in a fastfall-able state
     int state = hmn_data->state_id;
     //if ((state == ASID_JUMPF) || (state == ASID_JUMPB) || (state == ASID_JUMPAERIALF) || (state == ASID_JUMPAERIALB) || (state == ASID_FALL) || (state == ASID_FALLAERIAL) || ((state >= ASID_ATTACKAIRN) && (state <= ASID_ATTACKAIRLW))
     {
-        if (hmn_data->phys.self_vel.Y < 0) // can i fastfall?
+        if (can_fastfall)
         {
             // did i fastfall yet?
             if (hmn_data->flags.is_fastfall)
                 event_data->is_fastfall = 1; // set as fastfall this session
```

Comment:
This should be a bool and be set to true/false if it's semantically acting as a boolean value. In C it's all the same, but it helps for code clarity.

Reply:
Ok I'm reading the code and this issue is prevalent in many places in UP's original code. It might be outside the scope of this PR to fix all these issues. If you want to fix that in a later PR or leave for somebody else, that's fine.